lua/core/project.lua

```diff
@@ -1,11 +1,13 @@
 local M = {}
---
+
 function M.config()
   lvim.builtin.project = {
     --- This is on by default since it's currently the expected behavior.
     ---@usage set to false to disable project.nvim.
     active = true,
 
+    on_config_done = nil,
```

Comment:
can we separate active and on_config_done from the actual opt that we give to the `project.setup`

```lua
lvim.builtin.project = {
  active= true
  on_config_done = nil
  opts = {
   -- the rest
  },
```
so that we don't end up giving this whole table to
```lua
project.setup(settings)
```
a couple of lines down here

Reply:
I agree that this needs to be done, I was planing on doing it in a separate PR.
Also, the renaming of setup to config should probably be another one.
Can we keep a small scope for this one, as it's been open for a while now ?

Reply:
yeah but i suspect the issue in CI is bcz of one of these plugins not playing nice with `on_config_done` being given to them in setup

Reply:
I can take care of it then :+1: